conda/plugins.py

```diff
@@ -0,0 +1,31 @@
+from collections.abc import Iterable
+from typing import Callable, List, NamedTuple, Optional, Tuple
+
+import pluggy
+
+
+_hookspec = pluggy.HookspecMarker('conda')
+hookimp = pluggy.HookimplMarker('conda')
+
+
+class CondaSubcommand(NamedTuple):
```

Comment:
I know this would be a big dependency, but should we use attrs for this? Maybe something that would be useful elsewhere in the code base?

Reply:
I have been using `pydantic` on `menuinst`, if that helps. It's still not released or even merged, so we can change it too, but that's more of a CEP decision I am guessing? What for sure is not happening is us adding both `pydantic` and `attrs` to the dependency tree.

Reply:
Yeah, that's fair, both aren't great in terms of bus factor, so I'd say we should try to go with the one that has the most likely chance to be sustainable in the next years.

Reply:
Does conda vendor its dependencies? I don't think so, right? Even if it does, bringing new dependencies is problematic as lots of downstreams are required to unvendor those dependencies to comply to guidelines when packaging Conda. I'd highly recommend avoiding external dependencies unless absolutely necessary. A namedtuple is perfectly suitable for this use-case.
IMO this snippet should not motivate bringing any dependency. If we do decide bringing to bring in `attrs` or something like that, we could update it to use it, though even in that case I'd be inclined to not do that because this is public API and that would lock us into the dependency even if we decide to move away from it because it is no longer the best option for the code that motivated bringing it in the first place.

This would, of course, be a different story if the code did call for a functionality provided by a package like `attrs`, but in this case, it does not, and IMO that would only make the code more complex anyway :sweat_smile: